src/metabase/util.cljc

```diff
@@ -1038,6 +1039,45 @@
      [timer]
      (/ (- (System/nanoTime) timer) 1e6)))
 
+(defn group-by
+  "(group-by first second     [[1 3] [1 4] [2 5]]) => {1 [3 4], 2 [5]}
+   (group-by first second + 0 [[1 3] [1 4] [2 5]]) => {1 7,     2 5}
+   (group-by first second         [[1 [3]] [1 [4]] [2 [5]]]) => {1 [[3] [4]], 2 [[5]]}
+   (group-by first second concat  [[1 [3]] [1 [4]] [2 [5]]]) => {1 (3 4), 2 (5)}
+   (group-by first second into    [[1 [3]] [1 [4]] [2 [5]]]) => {1 [3 4], 2 [5]}
+   (group-by first second into [] [[1 [3]] [1 [4]] [2 [5]]]) => {1 [3 4], 2 [5]}
+   (group-by first second into () [[1 [3]] [1 [4]] [2 [5]]]) => {1 (4 3), 2 (5)}
+   ;; as filter:
+   (group-by first any? second even? conj () [[1 3] [1 4] [2 5]])      => {1 (4)}
+   ;; as reducer (see index-by below):
+   (group-by first any? second even? max   0 [[1 3] [1 6] [1 4] [2 5]] => {1 6})"
```

Comment:
Can we vertically align the example inputs and outputs here? This is hard to read.

Reply:
Also it should include an example for the 2-arity, eg. equivalent to OG `clojure.core/group-by`. I thought it was missing at first glance!